### ISSUE TITLE
Bug Report: async_hooks Resolution Error in Vercel CLI 39.1.2

### ISSUE THREAD

**Issue by iaaly** [zeit/now#12669]
## Description
The latest Vercel CLI version 39.1.2 fails to build on Cloudflare Pages due to an unresolved async_hooks dependency. The error occurs during the build process when trying to resolve this Node.js built-in module.

## Steps to Reproduce
1. Use Vercel CLI version 39.1.2
2. Deploy to Cloudflare Pages
3. Run build command

## Environment
Vercel CLI: 39.1.2
Node.js: v22.11.0
Deployment Platform: Cloudflare Pages
Build Command: npx vercel build

## Expected Behavior
Build should complete successfully as it did in version 39.1.1

## Current Behavior
Build fails with the following error:
```bash

15:37:33.427 | ▲  Build Completed in .vercel/output [51s]
-- | --
15:37:33.498 | ⚡️ Completed `npx vercel build`.
15:37:38.745 | ✘ [ERROR] Could not resolve "async_hooks"
15:37:38.745 |  
15:37:38.746 | <stdin>:388:680:
15:37:38.746 | 388 │ ...:()=>N});var R=A(require("async_hooks")),S="@next/request-cont...
15:37:38.746 | ╵                             ~~~~~~~~~~~~~
15:37:38.746 |  
15:37:38.747 | The package "async_hooks" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.
15:37:38.747 |  
15:37:38.754 | ✘ [ERROR] Could not resolve "async_hooks"
15:37:38.754 |  
15:37:38.754 | <stdin>:398:680:
15:37:38.754 | 398 │ ...:()=>N});var R=A(require("async_hooks")),S="@next/request-cont...
15:37:38.754 | ╵                             ~~~~~~~~~~~~~
15:37:38.754 |  
15:37:38.754 | The package "async_hooks" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.
15:37:38.755 |  
15:37:39.085 | /opt/buildhome/.npm/_npx/ae7d8149b8ab8d52/node_modules/esbuild/lib/main.js:1575
15:37:39.085 | let error = new Error(`${text}${summary}`);
15:37:39.085 | ^
15:37:39.085 |  
15:37:39.085 | Error: Build failed with 1 error:
15:37:39.086 | <stdin>:388:680: ERROR: Could not resolve "async_hooks"
15:37:39.086 | at failureErrorWithLog (/opt/buildhome/.npm/_npx/ae7d8149b8ab8d52/node_modules/esbuild/lib/main.js:1575:15)
15:37:39.086 | at /opt/buildhome/.npm/_npx/ae7d8149b8ab8d52/node_modules/esbuild/lib/main.js:1033:28
15:37:39.086 | at runOnEndCallbacks (/opt/buildhome/.npm/_npx/ae7d8149b8ab8d52/node_modules/esbuild/lib/main.js:1447:61)
15:37:39.086 | at buildResponseToResult (/opt/buildhome/.npm/_npx/ae7d8149b8ab8d52/node_modules/esbuild/lib/main.js:1031:7)
15:37:39.086 | at /opt/buildhome/.npm/_npx/ae7d8149b8ab8d52/node_modules/esbuild/lib/main.js:1143:14
15:37:39.086 | at responseCallbacks.<computed> (/opt/buildhome/.npm/_npx/ae7d8149b8ab8d52/node_modules/esbuild/lib/main.js:680:9)
15:37:39.086 | at handleIncomingPacket (/opt/buildhome/.npm/_npx/ae7d8149b8ab8d52/node_modules/esbuild/lib/main.js:735:9)
15:37:39.086 | at Socket.readFromStdout (/opt/buildhome/.npm/_npx/ae7d8149b8ab8d52/node_modules/esbuild/lib/main.js:656:7)
15:37:39.086 | at Socket.emit (node:events:518:28)
15:37:39.087 | at addChunk (node:internal/streams/readable:561:12) {
15:37:39.087 | errors: [
15:37:39.087 | {
15:37:39.087 | detail: undefined,
15:37:39.087 | id: '',
15:37:39.087 | location: {
15:37:39.087 | column: 680,
15:37:39.087 | file: '<stdin>',
15:37:39.087 | length: 13,
15:37:39.087 | line: 388,
15:37:39.088 | lineText: '    const fn = (function(module,exports) {var b=Object.create;var p=Object.defineProperty;var w=Object.getOwnPropertyDescriptor;var T=Object.getOwnPropertyNames;var P=Object.getPrototypeOf,L=Object.prototype.hasOwnProperty;var h=e=>p(e,"__esModule",{value:!0});var _=(e,n)=>{h(e);for(var t in n)p(e,t,{get:n[t],enumerable:!0})},U=(e,n,t)=>{if(n&&typeof n=="object"\|\|typeof n=="function")for(let i of T(n))!L.call(e,i)&&i!=="default"&&p(e,i,{get:()=>n[i],enumerable:!(t=w(n,i))\|\|t.enumerable});return e},A=e=>U(h(p(e!=null?b(P(e)):{},"default",e&&e.__esModule&&"default"in e?{get:()=>e.default,enumerable:!0}:{value:e,enumerable:!0})),e);_(exports,{default:()=>N});var R=A(require("async_hooks")),S="@next/request-context",f=Symbol.for(S),C=Symbol.for("internal.storage");function O(){let e=globalThis;if(!e[f]){let n=new R.AsyncLocalStorage,t={get:()=>n.getStore(),[C]:n};e[f]=t}return e[f]}var q=O();function m(e,n){return q[C].run(e,n)}function y(e){let n={};return e&&e.forEach((t,i)=>{n[i]=t,i.toLowerCase()==="set-cookie"&&(n[i]=M(t))}),n}function M(e){let n=[],t=0,i,a,g,o,r;function x(){for(;t<e.length&&/\\s/.test(e.charAt(t));)t+=1;return t<e.length}function s(){return a=e.charAt(t),a!=="="&&a!==";"&&a!==","}for(;t<e.length;){for(i=t,r=!1;x();)if(a=e.charAt(t),a===","){for(g=t,t+=1,x(),o=t;t<e.length&&s();)t+=1;t<e.length&&e.charAt(t)==="="?(r=!0,t=o,n.push(e.substring(i,g)),i=t):t=g+1}else t+=1;(!r\|\|t>=e.length)&&n.push(e.substring(i,e.length))}return n}function N(e){let n=e.staticRoutes.map(i=>({regexp:new RegExp(i.namedRegex),page:i.page})),t=e.dynamicRoutes?.map(i=>({regexp:new RegExp(i.namedRegex),page:i.page}))\|\|[];return async function(a,g){let o=new URL(a.url).pathname,r={};if(e.nextConfig?.basePath&&o.startsWith(e.nextConfig.basePath)&&(o=o.replace(e.nextConfig.basePath,"")\|\|"/"),e.nextConfig?.i18n)for(let s of e.nextConfig.i18n.locales){let u=new RegExp(`^/${s}($\|/)`,"i");if(o.match(u)){o=o.replace(u,"/")\|\|"/";break}}for(let s of n)if(s.regexp.exec(o)){r.name=s.page;break}if(!r.name){let s=E(o);for(let u of t\|\|[]){if(s&&!E(u.page))continue;let d=u.regexp.exec(o);if(d){r={name:u.page,params:d.groups};break}}}let x=await m({waitUntil:g.waitUntil},()=>globalThis._ENTRIES[`middleware_${e.name}`].default.call({},{request:{url:a.url,method:a.method,headers:y(a.headers),ip:c(a.headers,l.Ip),geo:{city:c(a.headers,l.City,!0),country:c(a.headers,l.Country,!0),latitude:c(a.headers,l.Latitude),longitude:c(a.headers,l.Longitude),region:c(a.headers,l.Region,!0)},nextConfig:e.nextConfig,page:r,body:a.body}}));return x.waitUntil&&g.waitUntil(x.waitUntil),x.response}}function c(e,n,t=!1){let i=e.get(n)\|\|void 0;return t&&i?decodeURIComponent(i):i}function E(e){return e==="/api"\|\|e.startsWith("/api/")}var l;(function(o){o.City="x-vercel-ip-city",o.Country="x-vercel-ip-country",o.Ip="x-real-ip",o.Latitude="x-vercel-ip-latitude",o.Longitude="x-vercel-ip-longitude",o.Region="x-vercel-ip-country-region"})(l\|\|(l={}));',
15:37:39.088 | namespace: '',
15:37:39.088 | suggestion: ''
15:37:39.088 | },
15:37:39.088 | notes: [
15:37:39.088 | {
15:37:39.088 | location: null,
15:37:39.088 | text: `The package "async_hooks" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.`
15:37:39.088 | }
15:37:39.088 | ],
15:37:39.089 | pluginName: '',
15:37:39.089 | text: 'Could not resolve "async_hooks"'
15:37:39.089 | }
15:37:39.089 | ],
15:37:39.089 | warnings: []
15:37:39.089 | }
```

**Comment by RobinDaugherty** [comment]
For reference, the discussion of the issue is at https://github.com/cloudflare/next-on-pages/issues/908

**Comment by vinayraghu1** [comment]
`cloudflare/next-on-pages` does not pin dependency version. It downloads latest Vercel. Everytime a new Vercel package is published, it breaks our Cloudflare deploys. I wish either Vercel changes didn't break stuff or Cloudflare would pin the versions.

**Comment by iaaly** [comment]
For anyone having this issue, you can temporary fix the version by adding this vercel version to your project, Cloudflare would then use it without fetching a newer version:

```bash
npm install -D vercel@39.1.1

**Comment by johan-windahl** [comment]
Solved with vercel@39.1.3 👍

**Comment by trek** [comment]
Glad to hear it @johan-windahl
